File: src/main/java/com/skyspaceoneoneone/service/ServiceRequestService.java
package com.skyspaceoneoneone.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspaceoneoneone.domain.ServiceRequest;
import com.skyspaceoneoneone.dto.ServiceRequestDTO;
import com.skyspaceoneoneone.dto.ServiceRequestSearchDTO;
import com.skyspaceoneoneone.dto.ServiceRequestPageDTO;
import com.skyspaceoneoneone.dto.ServiceRequestConvertCriteriaDTO;
import com.skyspaceoneoneone.service.GenericService;
import com.skyspaceoneoneone.dto.common.RequestDTO;
import com.skyspaceoneoneone.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ServiceRequestService extends GenericService<ServiceRequest, Integer> {

	List<ServiceRequest> findAll();

	ResultDTO addServiceRequest(ServiceRequestDTO serviceRequestDTO, RequestDTO requestDTO);

	ResultDTO updateServiceRequest(ServiceRequestDTO serviceRequestDTO, RequestDTO requestDTO);

    Page<ServiceRequest> getAllServiceRequests(Pageable pageable);

    Page<ServiceRequest> getAllServiceRequests(Specification<ServiceRequest> spec, Pageable pageable);

	ResponseEntity<ServiceRequestPageDTO> getServiceRequests(ServiceRequestSearchDTO serviceRequestSearchDTO);
	
	List<ServiceRequestDTO> convertServiceRequestsToServiceRequestDTOs(List<ServiceRequest> serviceRequests, ServiceRequestConvertCriteriaDTO convertCriteria);

	ServiceRequestDTO getServiceRequestDTOById(Integer serviceRequestId);







}

File: skyspaceoneoneone_ui/src/services/ServiceRequestService.js
import http from "../http-common"; 

class ServiceRequestService {
  getAllServiceRequests(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/serviceRequest/serviceRequests`, searchDTO);
  }

  get(serviceRequestId) {
    return this.getRequest(`/serviceRequest/${serviceRequestId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/serviceRequest?field=${matchData}`, null);
  }

  addServiceRequest(data) {
    return http.post("/serviceRequest/addServiceRequest", data);
  }

  update(data) {
  	return http.post("/serviceRequest/updateServiceRequest", data);
  }
  
  uploadImage(data,serviceRequestId) {
  	return http.postForm("/serviceRequest/uploadImage/"+serviceRequestId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ServiceRequestService();
